test/HarpoonBattleTest.cpp

```diff
@@ -0,0 +1,167 @@
+   auto error = std::string("no error");
+   auto merged = KrakenBattle::MergeData(uuid, type, ignored, error);
+
+   LOG(INFO) << "start here";
+   auto extracted = HarpoonBattle::ExtractToParts(merged);
+   EXPECT_EQ(std::get<HarpoonBattle::IndexOfSession>(extracted), uuid);
+
+   auto extractedType = std::get<HarpoonBattle::IndexOfReceivedType>(extracted);
+   EXPECT_EQ(extractedType, HarpoonBattle::ReceivedType::Done);
+   LOG(INFO) << "PAST THIS";
+
+   auto extractedChunk = std::get<HarpoonBattle::IndexOfChunk>(extracted);
+   auto dataAsString = KrakenIntegrationHelper::vectorToString(ignored);
+   auto extractedChunkAsString = KrakenIntegrationHelper::vectorToString(extractedChunk);
+   EXPECT_TRUE(dataAsString != extractedChunkAsString) << "data: " << dataAsString.size() << ",  extracted: " << extractedChunkAsString.size();
+   EXPECT_EQ(extractedChunkAsString.size(), 0);
+}
+
+
+TEST_F(HarpoonBattleTest, ChunksExtracted_END) {
+   const std::string uuid = "some-uuid";
+   const std::string expectedUuid = "00000000-0000-0000-0000-000000000000";
+   auto type = KrakenBattle::SendType::End;
+   Kraken::Chunks ignored;
+   auto error = std::string("no error");
+   auto merged = KrakenBattle::MergeData(uuid, type, ignored, error);
+
+   auto extracted = HarpoonBattle::ExtractToParts(merged);
+   std::string extractedUuid = std::get<HarpoonBattle::IndexOfSession>(extracted);
+   EXPECT_EQ(expectedUuid, expectedUuid);
+
+   auto extractedType = std::get<HarpoonBattle::IndexOfReceivedType>(extracted);
+   EXPECT_EQ(extractedType, HarpoonBattle::ReceivedType::End);
+
+   auto extractedChunk = std::get<HarpoonBattle::IndexOfChunk>(extracted);
+   auto dataAsString = KrakenIntegrationHelper::vectorToString(ignored);
+   auto extractedChunkAsString = KrakenIntegrationHelper::vectorToString(extractedChunk);
+   EXPECT_TRUE(dataAsString == extractedChunkAsString) << "data: " << dataAsString.size() << ",  extracted: " << extractedChunkAsString.size();
+   EXPECT_EQ(extractedChunkAsString.size(), 0);
+}
+
+
+
+TEST_F(HarpoonBattleTest, ChunksExtracted_ERROR) {
+   const std::string uuid = "some-uuid";
+   auto type = KrakenBattle::SendType::Error;
+   Kraken::Chunks ignored;
+   auto error = std::string("no error");
+   auto merged = KrakenBattle::MergeData(uuid, type, ignored, error);
+
+   auto extracted = HarpoonBattle::ExtractToParts(merged);
+   EXPECT_EQ(std::get<HarpoonBattle::IndexOfSession>(extracted), uuid);
+
+   
+   auto extractedType = std::get<HarpoonBattle::IndexOfReceivedType>(extracted);
+   EXPECT_EQ(extractedType, HarpoonBattle::ReceivedType::Error);
+
+   auto extractedChunk = std::get<HarpoonBattle::IndexOfChunk>(extracted);
+   auto dataAsString = KrakenIntegrationHelper::vectorToString(ignored);
+   auto extractedChunkAsString = KrakenIntegrationHelper::vectorToString(extractedChunk);
+   EXPECT_TRUE(dataAsString == extractedChunkAsString) << "data: " << dataAsString.size() << ",  extracted: " << extractedChunkAsString.size();
```

Comment:
I was under the impression the error message would get sent through. Shouldn't it be received here?